ray-operator/controllers/ray/common/pod.go

```diff
@@ -535,19 +535,25 @@ func labelPod(rayNodeType rayv1.RayNodeType, rayClusterName string, groupName st
 	}
 
 	for k, v := range ret {
-		if k == string(rayNodeType) {
-			// overriding invalid values for this label
-			if v != string(rayv1.HeadNode) && v != string(rayv1.WorkerNode) {
-				labels[k] = v
-			}
+		if _, ok := labels[k]; !ok {
+			labels[k] = v
```

Comment:
I don't think it is a good idea to modify the function argument `labels`. It's better to modify `ret` and then return it.